connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -269,7 +269,7 @@ public Response restartConnector(final @PathParam("connector") String connector,
             FutureCallback<Void> cb = new FutureCallback<>();
             herder.restartConnector(connector, cb);
             completeOrForwardRequest(cb, forwardingPath, "POST", headers, null, forward);
-            return Response.ok().build();
+            return Response.noContent().build();
```

Comment:
This returns `204 No Content`, right? But doesn't the old code (prior to this feature) return `200 OK` with no content? 

[KIP-745](https://cwiki.apache.org/confluence/display/KAFKA/KIP-745%3A+Connect+API+to+restart+connector+and+tasks) mentions using `202 Accepted` for the new behavior, but keeps the `200 OK` behavior when `includeTasks=false` and `onlyFailed=false`. 

Are you suggesting that we can't return `200 OK` here if there is no response body and should therefore update the KIP, or can this line be left as is and we change the `RestClient` to properly handle a null content? The issue for this test failure ([KAFKA-111]()) mentions [this code](https://github.com/apache/kafka/blob/d89ea74918ce0a1f2309a09473c9500c52af3b10/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java#L134-L136) in the `RestClient`, which handles all 200 to <300 status codes the same way, so could we just improve that?

Reply:
Thanks for the review @rhauch. I had the same questions and concerns. 

Actually, the current behavior for both connector and task restarts is to return `HTTP/1.1 204 No Content`. So this fix should retain this behavior for the connector restart. 

This seemed the most appropriate thing to do for a POST request who's response would have an empty body given the description in the RFC:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.2.5

I also think that an addendum to the KIP would be a good idea one we finish the review here. 
Wdyt?

Reply:
@rhauch  I confirme that from AK2.8 test the old behavior was returning 204, Please see below

> ➜  kafka_2.13-2.8.0 curl -v -X POST http://localhost:8083/connectors/local-file-source/restart
> *   Trying ::1...
> * TCP_NODELAY set
> * Connected to localhost (::1) port 8083 (#0)
> > POST /connectors/local-file-source/restart HTTP/1.1
> > Host: localhost:8083
> > User-Agent: curl/7.64.1
> > Accept: */*
> > 
> < HTTP/1.1 204 No Content
> < Date: Mon, 26 Jul 2021 23:41:44 GMT
> < Server: Jetty(9.4.39.v20210325)
> < 
> * Connection #0 to host localhost left intact
> * Closing connection 0
> ➜  kafka_2.13-2.8.0

Reply:
Awesome. Thanks, folks!

Reply:
I've corrected the KIP and sent an email describing this minor correction to the vote thread for the KIP.

I've also added [a comment on KAFKA-13139](https://issues.apache.org/jira/browse/KAFKA-13139?focusedCommentId=17388116&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17388116) that describes the root cause and the KIP correction.

Thanks, @kkonstantine and @kpatelatwork.